shared/profile/wot-author/container.tsx

```diff
@@ -1,53 +1,93 @@
-      )
+  const {username: trackerUsername, assertions} = Container.useSelector(state =>
+    Tracker2Constants.getDetails(state, voucheeUsername)
+  )
+  let proofs: Proof[] = []
+  if (trackerUsername === voucheeUsername) {
+    if (assertions) {
+      proofs = Array.from(assertions, ([_, assertion]) => assertion).filter(x => x.type !== 'stellar')
     }
-    return (
-      <Question1
-        error={error}
-        initialVerificationType={initialVerificationType}
-        onSubmit={onSubmit}
-        proofs={proofs}
-        voucheeUsername={voucheeUsername}
-      />
-    )
   } else {
-    const onSubmit = answer => {
-      console.log(answer) // PICNIC-1087 TODO use values
-      dispatch(RouteTreeGen.createClearModals())
+    error = `Proofs not loaded: ${trackerUsername} != ${voucheeUsername}`
+  }
+  const onSubmit = (answer: Question1Answer) => {
+    dispatch(
+      nav.safeNavigateAppendPayload({
+        path: [{props: {question1Answer: answer, username: voucheeUsername}, selected: 'profileWotAuthorQ2'}],
+      })
+    )
+  }
+  return (
+    <Question1
+      error={error}
+      initialVerificationType={'in_person'}
+      onSubmit={onSubmit}
+      proofs={proofs}
+      voucheeUsername={voucheeUsername}
+    />
+  )
+}
+
+const Question2Wrapper = (
+  props: Container.RouteProps<{
+    username: string
+    question1Answer: Question1Answer
+  }>
+) => {
+  const voucheeUsername = Container.getRouteProps(props, 'username', '')
+  const question1Answer = Container.getRoutePropsOr(props, 'question1Answer', 'error')
+  const nav = Container.useSafeNavigation()
+  const dispatch = Container.useDispatch()
+  let error = Container.useSelector(state => state.profile.wotAuthorError)
+  if (!error && !voucheeUsername) {
+    error = 'Routing missing username.'
+  }
+  if (!error && question1Answer === 'error') {
+    error = 'Routing missing q1 answer.'
+  }
+  const waiting = Container.useAnyWaiting(Constants.wotAuthorWaitingKey)
+  const onSubmit = ({statement}: {statement: string}) => {
+    if (question1Answer === 'error') {
+      return
     }
-    return (
-      <Question2
-        error={error}
-        onBack={() => dispatch(nav.safeNavigateUpPayload())}
-        onSubmit={onSubmit}
-        voucheeUsername={voucheeUsername}
-      />
+    const {otherText, verificationType} = question1Answer
+    dispatch(
+      ProfileGen.createWotVouch({
+        otherText,
+        proofs: [], // PICNIC-1087 TODO
+        statement,
+        username: voucheeUsername,
+        verificationType,
+      })
     )
   }
+  const onBack = () => {
+    dispatch(ProfileGen.createWotVouchSetError({error: ''}))
+    dispatch(nav.safeNavigateUpPayload())
+  }
+  return (
+    <Question2
+      error={error}
+      onBack={onBack}
+      onSubmit={onSubmit}
+      voucheeUsername={voucheeUsername}
+      waiting={waiting}
+    />
+  )
 }
 
-export default WotAuthorRoot
+export {Question1Wrapper as Question1, Question2Wrapper as Question2}
```

Comment:
Move to hooks is good, however files named `container.tsx` imply a `react-redux` component - you know the `Container.connect(...)` thing.

I see why the split is helpful. I would strongly consider moving these wrapping components directly into `profile/wot-author/index.tsx`.

Also, with hooks, there's no need to split things into separate components. Your re-rendering is simplified if you move all hooks into one component. This of course has the downside that your components can get very long and bloated.

Reply:
Ok, will move them to the same file. It seemed like the conceptual division is the same even though hooks are used instead of connect. Far away callers import container and don't care whether it's called Wrapper or not. Now routes.tsx will call *Wrapper.

Divided components make stories easy. The wrappers connect to the store but the unwrappeds are dumb enough to get story props. I hear there's injector things but I don't know them, heard they may be broken on mobile, and stories don't need to care about redux state shape.